modin/core/dataframe/pandas/partitioning/partition.py

```diff
@@ -116,7 +122,9 @@ def to_pandas(self):
         If the underlying object is a pandas DataFrame, this will likely
         only need to call `get`.
         """
-        pass
+        dataframe = self.get()
+        assert isinstance(dataframe, (pandas.DataFrame, pandas.Series))
```

Comment:
Since a partition should hold a pandas DataFrame I think we can remove the check for a pandas Series.
cc @devin-petersohn